test/e2e/upgrade/adminack/adminack.go

```diff
@@ -47,6 +47,10 @@ func (t *UpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade
 	go func() {
 		<-done
 		cancel()
+		// Perform one guaranteed check after the upgrade is complete. We cancel the polled check
+		// above, so we never know whether the poll was lucky to run at least once since the version
+		// was bumped.
+		(&clusterversionoperator.AdminAckTest{Oc: t.oc, Config: t.config}).Test(context.Background())
```

Comment:
do we want to cap the amount of time we're willing to spend on this?   `context.WithTimeout(context.Background(), 5*time.Minute)` or similar?

Reply:
I considered that but I checked the other place that does this one-off check and it also just does `context.Background()`:

https://github.com/openshift/origin/blob/74d8b6dc8a1803fd6153e5abbe15f96e67cc763e/test/extended/adminack/adminack.go#L23-L26

But yeah, it's probably better to cap that. I'll try to come up with some reasonable upper bound - there seems to be multiple timeouted polls inside the check, potentially even iterated, the upper bound needs to match the expected sum of these.

Reply:
There are two 3 minute waits for each applicable gate. I don't think we ever had more than one gate but with a little future-proofing I set the timeout to 15m  to handle a potential case where we have two gates and somehowtthings go slowly enough to hit the three minute timeout on each wait (this is IMO totally pathological case).

Reply:
Huh I guess I cannot do the check here because `Test` will terminate while we're still testing in the goroutine.